exercises/registers/registers03.cairo

```diff
@@ -0,0 +1,101 @@
+%lang starknet
+
+# I AM NOT DONE
+
+# Cairo memory is immutable, which mean that once a memory cell have been assigned a value it be changed.
+# The program will crash if someone try to assign a new, different, value to a already initialized memory cell.
+# However, trying to assign twice, or more, a memory cell will the same value won't cause any harm.
+#
+# This property can be used to assert the value of a cell.
+# By trying to give it a value to an already initialized memory cell, we can ensure that it already have this specific value,
+# if it does not, the call fail
+
+# TODO
+# Rewrite this function in a high level syntax, using tempvar and assert
+func crash():
+    # [ap] = 42; ap++
+    # [ap - 1] = 42
+    # [ap - 1] = 21
+
+    ret
+end
+
+# TODO
+# Rewrite this funtion in a low level syntax
+func assert_42(number : felt):
+    # assert number = 42
+
+    return ()
+end
+
+# TODO
+# Write this function body so:
+# if the memory cell pointed by `p_number` is not initialised, it set it to 42
+# else if the value is initialized and different from 42, it crash
+# else, do nothing and return
+func assert_pointer_42(p_number : felt*):
+    return ()
+end
+
```

Comment:
Suggestion: Adding a counterpart exercise that "only assert"
By using the fact that `assert 42 = [pointer]` will never write to any memory cell and only assert (since 42 is a constant)

Reply:
done